core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -195,8 +203,108 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
     throw new UnsupportedOperationException("Cannot rename Hadoop tables");
   }
 
+  @Override
+  public boolean createNamespace(Namespace namespace) {
+    Preconditions.checkArgument(!namespace.isEmpty(), "Your Namespace must be not empty!");
+    Joiner slash = Joiner.on("/");
+    Path nsPath = new Path(slash.join(warehouseLocation, slash.join(namespace.levels())));
+    return io().mkdir(nsPath.toString());
+  }
+
+  @Override
+  public List<Namespace> listNamespaces() {
+    return listNamespaces(Namespace.empty());
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+ //   Preconditions.checkArgument(!namespace.isEmpty(), "Your Namespace must be not empty!");
+    List<Namespace> namespaces = new ArrayList<>();
+    String[] nsp;
+    Joiner slash = Joiner.on("/");
+    Path nsPath = new Path(slash.join(warehouseLocation, slash.join(namespace.levels())));
+    FileSystem fs = Util.getFs(nsPath, conf);
+    try {
+      if (!fs.exists(nsPath) || !fs.isDirectory(nsPath)) {
+        throw new NotFoundException("Unknown namespace " + namespace);
+      }
+
+      for (FileStatus s : fs.listStatus(nsPath)) {
+        Path path = s.getPath();
+        if (!fs.isDirectory(path)) {
+          // Ignore the path which is not a directory.
+          continue;
+        }
+        Path metadataPath = new Path(path, "metadata");
+        if (fs.exists(metadataPath) && fs.isDirectory(metadataPath) &&
+            (fs.listStatus(metadataPath, TABLE_FILTER).length >= 1)) {
+        // Ignore the path of table
+          continue;
+        }
+
+        if (fs.exists(path) && fs.isDirectory(path)) {
+          // Only the path which contains metadata is the path for table, otherwise it could be
+          // still a namespace.
+          if (!namespace.isEmpty()) {
+            nsp = (namespace.toString() + "." + path.getName()).split("\\.");
+          } else {
+            nsp = new String[] {path.getName()};
+          }
+          namespaces.add(Namespace.of(nsp));
+        }
+      }
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to list namespace under " + namespace, ioe);
+    }
+
+    return namespaces;
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) {
+    Preconditions.checkArgument(!namespace.isEmpty(), "Your Namespace must be not empty!");
+    Preconditions.checkArgument(listTables(namespace).size() == 0,
+        "This Namespace have tables, cannot drop it");
+    Preconditions.checkArgument(listNamespaces(namespace).size() == 0,
+        "This Namespace have sub Namespace, cannot drop it");
+
+    Joiner slash = Joiner.on("/");
+    Path nsPath = new Path(slash.join(warehouseLocation, slash.join(namespace.levels())));
+    io().deleteFile(nsPath.toString());
+    return true;
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) {
```

Comment:
I don't think that the Hadoop catalog needs to support namespace metadata. There is no good way to store it, so let's skip it for now.

Reply:
The SparkSQL call `loadNamespaceMetadata`  first while create namespace, so I think that  needs to support in `HadoopCatalog` and `HiveCatalog`

Reply:
This can return an empty map for Hadoop tables, since there is no namespace metadata. I think it is also okay to return "location" with the namespace's path in the map.